crates/autopilot/src/infra/solvers/dto/solve.rs

```diff
@@ -182,6 +182,37 @@ pub struct Solution {
     pub gas: Option<u64>,
 }
 
+fn deserialize_solution_id<'de, D>(deserializer: D) -> Result<u64, D::Error>
+where
+    D: serde::Deserializer<'de>,
+{
+    struct SolutionIdVisitor;
```

Comment:
Alternatively this can also be implemented using a helper enum. Might be a bit more self explanatory.
```rust
fn deserialize_solution_id<'de, D>(deserializer: D) -> Result<u64, D::Error>
where
    D: serde::Deserializer<'de>,
{
    #[serde_as]
    #[derive(Deserialize, Debug)]
    #[serde(untagged)]
    enum StringOrInt {
        String(#[serde_as(as = "DisplayFromStr")] u64),
        Int(u64)
    }

    let result = match StringOrInt::deserialize(deserializer)? {
        StringOrInt::String(id) => id,
        StringOrInt::Int(id) => id
    };

    Ok(result)
}
```

Reply:
This one is nice as well. But I guess it's ok keeping the current version since it will be removed in one week from now.